plugin/storage/kafka/options.go

```diff
@@ -94,3 +181,36 @@ func (opt *Options) InitFromViper(v *viper.Viper) {
 func stripWhiteSpace(str string) string {
 	return strings.Replace(str, " ", "", -1)
 }
+
+// getCompressionLevel to get compression level from compression type
+func getCompressionLevel(compressionMode string, compressionLevel int) (int, error) {
+	compressionModeData := compressionModes[compressionMode]
+
+	if compressionLevel == defaultCompressionLevel {
+		return compressionModeData.defaultCompressionLevel, nil
+	}
+
+	if compressionModeData.minCompressionLevel <= compressionLevel && compressionModeData.maxCompressionLevel >= compressionLevel {
```

Comment:
I suggest reversing the condition and returning an error "compression level is not within valid range"

Reply:
fixed